src/lib/seam/thermostats/use-update-thermostat.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  useMutation,
+  type UseMutationResult,
+  useQueryClient,
+} from '@tanstack/react-query'
+import type {
+  SeamError,
+  ThermostatDevice,
+  ThermostatsListResponse,
+  ThermostatUpdateRequest,
+} from 'seamapi'
+
+import { NullSeamClientError, useSeamClient } from 'lib/index.js'
+
+type UseUpdateThermostatData = Record<string, unknown>
+type UseUpdateThermostatMutationParams = ThermostatUpdateRequest
+
+export function useUpdateThermostat(): UseMutationResult<
+  UseUpdateThermostatData,
+  SeamError,
+  UseUpdateThermostatMutationParams
+> {
+  const { client } = useSeamClient()
+  const queryClient = useQueryClient()
+
+  return useMutation<
+    UseUpdateThermostatData,
+    SeamError,
+    UseUpdateThermostatMutationParams
+  >({
+    mutationFn: async (mutationParams: UseUpdateThermostatMutationParams) => {
+      if (client === null) throw new NullSeamClientError()
+
+      return await client.thermostats.update(mutationParams)
+    },
+    onSuccess: (_data, variables) => {
+      queryClient.setQueryData<ThermostatDevice | null>(
+        ['thermostats', 'get', { device_id: variables.device_id }],
+        (thermostat) => {
+          if (thermostat == null) {
+            return
+          }
+
+          return getUpdatedThermostat(thermostat, variables)
+        }
+      )
+
+      queryClient.setQueryData<ThermostatsListResponse['thermostats']>(
+        ['thermostats', 'list', { device_id: variables.device_id }],
+        (thermostats): ThermostatDevice[] => {
+          if (thermostats == null) {
+            return []
+          }
+
+          return thermostats.map((thermostat) => {
+            if (thermostat.device_id === variables.device_id) {
+              return getUpdatedThermostat(thermostat, variables)
+            }
+
+            return thermostat
+          })
+        }
+      )
+    },
+  })
+}
+
+function getUpdatedThermostat(
+  thermostat: ThermostatDevice,
+  variables: UseUpdateThermostatMutationParams
```

Comment:
Was this param meant to be used?

Reply:
`variables`? Yes, it is used below on line 87 (unless I misunderstood)

Reply:
Ah ok, my browser search highlighter didn't work right so I missed it.